src/celeritas/optical/OpticalPhysicsParams.hh

```diff
@@ -0,0 +1,129 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/OpticalPhysicsParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "corecel/Types.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionMirror.hh"
+#include "corecel/data/ParamsDataInterface.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/global/ActionRegistry.hh"
+
+#include "OpticalPhysicsData.hh"
+
+namespace celeritas
+{
+class OpticalPropertyParams;
+class OpticalModel;
+class OpticalModelBuilder;
+class ConcreteAction;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Options to construct the optical physics data.
+ */
+struct OpticalPhysicsParamsOptions
+{
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Parameter interface for optical physics data.
+ *
+ * Performs the construction of optical physics data and optical models
+ * from imported data, and provides an interface for accessing this data
+ * on the host and on the device.
```

Comment:
It would be good to add a few notes about how this class differs from the main EM physics:
- We interpolate linearly on MFP vs energy, rather than macroscopic XS vs log energy (or scaled log energy)
- There's a one-to-one model/process correlation
- Only a single incident particle type is supported
- Not all materials in the problem have optical data (?)
- ....

Reply:
>     * We interpolate linearly on MFP vs energy, rather than macroscopic XS vs log energy (or scaled log energy)

A thought on linear interpolation: this is effectively just due to our choice of grid, right? AFAIK, the code in `OpticalPhysicsParams` and determining the discrete action is agnostic of the grid, other than the fact it will calculate some XS / MFP for a given energy value. It's definitely important whether we're using MFP vs macroscopic XS, but otherwise the code behaves mostly the same as in the EM case when dealing with these grids.